clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7563,6 +7574,18 @@ static void handleSYCLAddIRAttributesFunctionAttr(Sema &S, Decl *D,
   S.AddSYCLAddIRAttributesFunctionAttr(D, A, Args);
 }
 
+SYCLAddIRAttributesKernelParameterAttr *
+Sema::MergeSYCLAddIRAttributesKernelParameterAttr(
+    Decl *D, const SYCLAddIRAttributesKernelParameterAttr &A) {
+  if (const auto *ExistingAttr =
+          D->getAttr<SYCLAddIRAttributesKernelParameterAttr>()) {
+    Diag(ExistingAttr->getLoc(), diag::warn_duplicate_attribute_exact) << &A;
+    Diag(A.getLoc(), diag::note_previous_attribute);
+    return nullptr;
+  }
```

Comment:
Same question?

Reply:
Same concerns here as above (and for the same reasons).

Reply:
See resolution https://github.com/intel/llvm/pull/5535#discussion_r823774609. All three attributes have the same requirements when merging.